src/server/system_services/bucket_server.js

```diff
@@ -76,6 +77,60 @@ function new_bucket_defaults(name, system_id, tiering_policy_id, owner_account_i
     };
 }
 
+function auto_setup_tier2(req, changes, skip_check = false) {
+    if (!config.BUCKET_AUTOCONF_TIER2_ENABLED) return;
+    if (!changes.insert) return;
+
+    const tiering_policies = changes.insert.tieringpolicies;
+    if (!tiering_policies) return;
+
+    const initial_tier = changes.insert.tiers?.[0];
+    if (!initial_tier) return;
```

Comment:
It's not bad, but how about sending the tier1 and tiering_policy objects as arguments instead of assuming their location in the changes object?

Reply:
I missed this comment actually